### PR TITLE
tailscale: Update to 1.48.0

### PR DESCRIPTION
**Pull request by mochaaP**
Maintainer: @ja-pa @oskarirauta
Compile tested: `aarch64_cortex-a53` @ SNAPSHOT
Run tested: `aarch64_cortex-a53` @ SNAPSHOT

This update brings nftables support to Tailscale.

Signed-off-by: Zephyr Lykos <git@mochaa.ws>

### CODE REVIEW

**Comment by BKPepe** [comment]
This really needs to be run tested as there are huge changes.

**Comment by mochaaP** [comment]
Agree that, but I temporarily have no access to my testing device. Feedbacks welcomed!

**Comment by mochaaP** [comment]
race cond in CI, please rerun

**Comment by BKPepe** [comment]
Still, to be merged, it requires to be run tested. I will not merge this unless there is confirmation or some indication of run testing. The changes are huge.

**Comment by cmtsij** [comment]
@mochaaP 
Do we have to set TS_DEBUG_USE_NETLINK_NFTABLES=true to enable nftables ?
ref: https://forum.openwrt.org/t/tailscale-with-nftables-support/166872

**Comment by mochaaP** [comment]
Run tested: `aarch64_cortex-a53` @ SNAPSHOT

Everything works including routing with nft.

Users who set up the rules with fw4 can try running with `--netfilter-mode=on` and check the rules with `nft list ruleset`.

**Comment by mochaaP** [comment]
cc @BKPepe

File: net/tailscale/patches/010-remove_iptables.patch
Comment (BKPepe) [inline net/tailscale/patches/010-remove_iptables.patch:1]: Ok, so... this patch... 

- Who is the author?
   - Who did those changes and when we can find them?
- Where is the commit subject and commit description for this patch?
- Why do we need this patch at all? Why it is not included in tailscale repository?
- Will you send it to upstream?

File: net/tailscale/Makefile
Comment (BKPepe) [inline net/tailscale/Makefile:14]: PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz

File: net/tailscale/patches/010-remove_iptables.patch
Comment (mochaaP) [reply net/tailscale/patches/010-remove_iptables.patch:1]: > Who is the author?

Me.

> Where is the commit subject and commit description for this patch?

It's in the main commit message: `the go-iptables support is no longer compiled in.`

> Why do we need this patch at all? Why it is not included in tailscale repository?

I removed iptables related code and ran `go mod tidy` on the dependencies. This eliminates the go-iptables dependency to reducing binary size and the need to add a flag for this.

> Will you send it to upstream?

No, as upstream will not accept those changes, and we only need this patch in the branches without iptables support.

File: net/tailscale/Makefile
Comment (mochaaP) [reply net/tailscale/Makefile]: resolved in <https://github.com/openwrt/packages/commit/aeedbf9c101e3c2fe653ab9f19a0855514b7bdcb>

File: net/tailscale/patches/010-remove_iptables.patch
Comment (BKPepe) [reply net/tailscale/patches/010-remove_iptables.patch:1]: Sorry, but no. This patch is so huge, then we can not accept it here. Nobody will look into it and refreshing or editing this patch will be pain.  You should reach upstream to check what needs to be done here.

Even it is in the commit message, it is not in the patch itself.

There is even iptables support in OpenWrt master branch. Not enabled by default, but still there is.

**Comment by 1715173329** [comment]
Ping @oskarirauta
Please take a look, thanks

File: net/tailscale/patches/010-remove_iptables.patch
Comment (mochaaP) [reply net/tailscale/patches/010-remove_iptables.patch:1]: But I don't see a point where we should maintain the old patch for go-iptables while we have support for nftables. The old fake_iptables patch is only required to workaround detections in go-iptables as it would check for the existence of the `iptables` binary on system.

The huge patch size is only because deleting files have to count every line of code and updating the indirect dependencies required by both go-iptables and other packages.

I could add a script to automate refreshing the patch if you wish me to.

:heart:

Comment (BKPepe) [reply net/tailscale/patches/010-remove_iptables.patch:1]: It was not removed upstream, why should we do that if others GNU/LInux distros did not do it as well? Nftables was added only 3 weeks ago and now it seems like you are eager to see iptables support drop, then discuss that it upstream, please.

Comment (mochaaP) [reply net/tailscale/patches/010-remove_iptables.patch:1]: I should outline the problem here:

**The old fake_iptables patch is only required to workaround detections in go-iptables as it would check for the existence of the iptables binary on system and refuse to run if none is provided.**

iptables support is already broken a while ago, as it does nothing but executes `/bin/false`. The "support" is just to make the program run and the routing functionality is not provided anyways. For me it doesn't make much sense to continue supporting the fake iptables patch, if the actual feature is working via nftables.

**Comment by mochaaP** [comment]
We finally don't need any patches on our side. :smiley:

cc @1715173329 @BKPepe

**Comment by mochaaP** [comment]
Squashed into a single commit as per Shen suggests.

**Review by 1715173329 — APPROVED** [review]
LGTM - Thanks!

**Pull request merged**
